pkg/syncer/spec/mutators/deployment_test.go

```diff
@@ -490,9 +578,31 @@ func TestMutate(t *testing.T) {
 	} {
 		{
 			t.Run(c.desc, func(t *testing.T) {
+				ctx, cancelFn := context.WithCancel(context.Background())
+				t.Cleanup(cancelFn)
+
 				upstreamURL, err := url.Parse(c.config.Host)
 				require.NoError(t, err)
-				dm := NewDeploymentMutator(upstreamURL)
+				fakeClient := fake.NewSimpleClientset(toObjects(c.upstreamSecrets)...)
+				sharedInformer := informers.NewSharedInformerFactory(fakeClient, 0)
+				informer, err := sharedInformer.ForResource(schema.GroupVersionResource{Version: "v1", Resource: "secrets"})
+				require.NoError(t, err)
+				indexer := informer.Informer().GetIndexer()
+				err = indexer.AddIndexers(cache.Indexers{
+					"by-workspace": func(obj interface{}) ([]string, error) {
```

Comment:
use the existing index func byWorkspace

Reply:
an idea: don't pass an index to the mutator, but pass a List func. That's is much easier to mock, no need for indexes and stuff and fake-clients.